outlines/models/vllm_server.py

```diff
@@ -0,0 +1,237 @@
+from typing import TYPE_CHECKING, Union
+
+from outlines.models.base import Model, ModelTypeAdapter
+from outlines.models.openai import OpenAITypeAdapter
+from outlines.types.dsl import python_types_to_terms, to_regex, JsonSchema, CFG
+
+
+if TYPE_CHECKING:
+    from openai import AsyncOpenAI, OpenAI
+
+
+class VLLMServerTypeAdapter(ModelTypeAdapter):
+
+    def format_input(self, model_input):
+        """Generate the prompt argument to pass to the client.
+
+        We rely on the OpenAITypeAdapter to format the input as the vLLM server
+        expects input in the same format as OpenAI.
+
+        Argument
+        --------
+        model_input
+            The input passed by the user.
+
+        """
+        return OpenAITypeAdapter().format_input(model_input)
+
+    def format_output_type(self, output_type):
+        """Generate the structured output argument to pass to the client.
+
+        Argument
+        --------
+        output_type
+            The structured output type provided.
+
+        """
+        if output_type is None:
+            return {}
+
+        term = python_types_to_terms(output_type)
+        if isinstance(term, CFG):
+            return {"guided_grammar": term.definition}
+        elif isinstance(term, JsonSchema):
+            return {"guided_json": term.schema, "whitespace_pattern": term.whitespace_pattern}
+        else:
```

Comment:
Shouldn't there be a `try` / `else` here?

Reply:
The functions `python_types_to_terms` and `to_regex` already include error messages in case of unsupported types. What do you want to put in the `except` block?